### PR TITLE
154 improve quiz journey

### PR DESCRIPTION
**Pull request by rosejbon**
https://trello.com/c/0UznraUp/154-improve-journey-from-when-people-clicked-a-negative-emotion-to-back-to-the-quiz-vv-ee

### CODE REVIEW

**Comment by marxian** [comment]
#### 📝Changed routes:


| [**/quiz/helplines**](https://my-best-life-lnc781yqa-neontribe.vercel.app/quiz/helplines) |
|:-:|
|<a href="https://my-best-life-lnc781yqa-neontribe.vercel.app/quiz/helplines">Screenshot of /quiz/helplines</a><br /><sup><a href="https://images.weserv.nl?url=https://iad.microlink.io/0IzfqV5b9m4Kt1R0FUECdh7KE_oZ8Mf1rSuHgbyuEixK0j1J3uFdTZ8jytM5WkxlOy6sdRHNjxdaNfxz4BEE6w.png">(view full size)</a> |







Commit c3e3366122383cec247503f1d15245f1a9d90dee (https://my-best-life-lnc781yqa-neontribe.vercel.app).

File: pages/quiz/helplines.tsx [hunk 1]
```diff
@@ -65,27 +94,78 @@ const StyledLink = styled.a`
...
+
+    for (let i = 0; i < feelings.length; i++) {
+      if (triggerFeelings.find((element) => element === feelings[i])) {
+        newTriggerFeelings.push(feelings[i])
+        setSelectedTriggerFeelings(newTriggerFeelings)
+      }
+    }
+  }, [])
+
+  const makeFeelingsList = (array: string[]) => {
```
Comment (edpe) [inline pages/quiz/helplines.tsx:121]: I really like this bit, I think it's really clever

File: pages/quiz/helplines.tsx [hunk 2]
```diff
@@ -65,27 +94,78 @@ const StyledLink = styled.a`
...
-        <Text>If you are in immediate danger, call 999.</Text>
+        <MainText>{`You've selected that you feel ${makeFeelingsList(
+          selectedTriggerFeelings
+        )}`}</MainText>
+        <VerticalSpacing size={2} />
+        <MainText>
+          Do you need help now or you are just having a bad day?
+        </MainText>
+        <VerticalSpacing size={2} />
+        <Link href="/quiz/what-are-your-interests" passHref>
```
Comment (edpe) [inline pages/quiz/helplines.tsx:140]: Not sure that this amount of text works here, maybe full width and give it some more padding or make it a link and keep text-decoration on?

File: pages/quiz/helplines.tsx [hunk 3]
```diff
@@ -94,3 +174,14 @@ export const IfYouNeedHelpPage = (): JSX.Element => {
 }
 
 export default IfYouNeedHelpPage
+
+export const getStaticProps: GetStaticProps = async () => {
```
Comment (edpe) [inline pages/quiz/helplines.tsx:178]: You got it all working, nice! :1st_place_medal:

File: pages/quiz/helplines.tsx [hunk 4]
```diff
@@ -65,27 +94,78 @@ const StyledLink = styled.a`
...
+        <MainText>{`You've selected that you feel ${makeFeelingsList(
+          selectedTriggerFeelings
+        )}`}</MainText>
+        <VerticalSpacing size={2} />
+        <MainText>
+          Do you need help now or you are just having a bad day?
+        </MainText>
+        <VerticalSpacing size={2} />
+        <Link href="/quiz/what-are-your-interests" passHref>
+          <StyledLink>{'A bad day, carry on with the quiz'}</StyledLink>
```
Comment (edpe) [inline pages/quiz/helplines.tsx:141]: Not sure that this amount of text works here, maybe full width and give it some more padding or make it a link and keep text-decoration on?

File: pages/quiz/helplines.tsx [hunk 5]
```diff
@@ -65,27 +94,78 @@ const StyledLink = styled.a`
...
+      if (triggerFeelings.find((element) => element === feelings[i])) {
+        newTriggerFeelings.push(feelings[i])
+        setSelectedTriggerFeelings(newTriggerFeelings)
+      }
+    }
+  }, [])
+
+  const makeFeelingsList = (array: string[]) => {
+    if (array.length === 1) return array[0]
+    const firstFeelings = array.slice(0, array.length - 1)
```
Comment (edpe) [inline pages/quiz/helplines.tsx:123]: I really like this bit, I think it's really clever

File: pages/quiz/helplines.tsx (hunk 4 above)
Comment (rosejbon) [reply pages/quiz/helplines.tsx:141]: Rachel is going to have a think and get back with design improvements to make this page a bit nicer. I agree it doesn't quite work right now.

**Review by Nikomus — APPROVED** [review]
Let's go for it - I think you'll want to pull out the HeaderComponent and the StickyNavBar in the merge conflict 🚁

**Comment by rosejbon** [comment]
> Let's go for it - I think you'll want to pull out the HeaderComponent and the StickyNavBar in the merge confli

I've made some changes based on Rachel's designs.

File: pages/quiz/helplines.tsx [hunk 6]
```diff
@@ -61,32 +126,171 @@ const StyledLink = styled.a`
...
+          serviceData2={serviceData2}
+        />
+      </MainBody>
 
-      <Link href="/quiz/what-are-your-interests" passHref>
-        <StyledLink>{'Back to the quiz'}</StyledLink>
-      </Link>
+      <VerticalSpacing size={4} />
 
       <VerticalSpacing />
```
Comment (Nikomus) [inline pages/quiz/helplines.tsx:280]: Doubled VerticalSpacing?

Comment (rosejbon) [reply pages/quiz/helplines.tsx]: Good spot, I've removed

**Review by Nikomus — APPROVED** [review]
Let's go for it! 🌯

**Pull request merged**